src/operations/get-patch-entity.handler.ts

```diff
@@ -14,6 +14,15 @@ export const getPatchEntityHandler = (methodConfigs: PatchOperationConfig, dataA
     const patchedItem = dataAdapter.patchOne(paramsFilter, req.body, methodConfigs.save);
 
     await delay(methodConfigs.delay)
-    methodConfigs.returnEntity ? res.send(patchedItem) : res.end();
+    if (methodConfigs.returnEntity) {
+      if (methodConfigs.responseModel?.paths?.data) {
+        const response = getResponseModel(patchedItem, methodConfigs.responseModel.paths.data);
+
+        res.send(response);
+      }
+      return;
```

Comment:
my suggestion is to create a class/factory for a `responseBuilder` which gets `responseModel` and then on each step you give it some info like `builder.writeData(patchedItem)` and it writes it and later we can add a method builder.writeTotalCount(count) and it'll write it too.